### ISSUE TITLE
Duplicate choices appear for some "Goto Definition" results

### ISSUE THREAD

**Issue by asfaltboy** [DamnWidget/anaconda#497]
This may be related to having some packages installed in "editable" mode (I have multiple such packages in my env).

**Comment by DamnWidget** [comment]
Is the double choice present for every symbol or is for some specific libraries? The Goto, Autocomplete and Show Documentation features are provided by Jedi so I guess it is returning those dupes.

Can you try to run jedi manually in an IPython (or similar) from the anaconda's `anaconda_lib` directory using a symbol that gave you the problem?

For example, and using the file `test.py` as sample:
```python
import jedi
with open('~/projects/my_project/test.py',  'r') as f:
    jedi.Script(f.read()).completions()
```

Note that the last character of the file **must** be in a letter of the symbol word that gave you dupes, you can also use a multiline string using `'''` if you are more comfortable.

**Comment by asfaltboy** [comment]
Yes, good call this is indeed an issue with jedi. Seems `goto_definitions` computes a `sys.path` (when not provided) and includes the editable package source path twice. I'll open an issue with `jedi`.

**Comment by asfaltboy** [comment]
I was dead wrong, it's not dupe at all! It's just defined multiple times...

**Comment by DamnWidget** [comment]
That is life ;)

**Comment by asfaltboy** [comment]
Sorry, I have to raise this again. I confused two scenarios. There still is a case, where I get 2 identical locations when using "Anaconda: Goto object definition" command.

I'm not sure exactly how to reproduce this in jedi shell, but it is plausible the cause is multiple definitions returned for a given `completion.follow_definition()`. It is this completion object's method that I'm attempting to follow.
The dupe definition returned seems to be caused by having a conditional import statement (try import OneClass, except import AnotherClass), where both classes imported are eventually initiated and assigned to same symbol in the module.

Regardless, I now feel my first instinct was correct, I think Anaconda should consider this (entirely legitimate) scenario and remove any dupe entries. I would consider duplicates entries that have all of the following exactly equal:

  * same path string
  * same line and same row

@DamnWidget what do you say?

**Comment by DamnWidget** [comment]
I am not sure if I want to add that overhead to every single request, look for dupes means add in a way or another an O(n^2) operation that is kinda slow to solve a problem that I consider to be minor.

**Comment by asfaltboy** [comment]
Yes, in theory, but,we're talking about a lookup for `len(matches) > 1`, I can't imagine that will happen often, and if it does it will be 3 conditions checked, maybe a single digit number of times ... I wouldn't call that performance overhead, only code complexity.

**Comment by DamnWidget** [comment]
I was taking a look at the code and at your solution, I used the `os.path` as sample and I don't think your solution is valid as in the multiple import scenario the line and column are different, you can check it running the following:
```python
import jedi
data = [(i.module_path, i.line, i.column + 1) for i in jedi.Script('import os; os.path').goto_assigments() if not i.in_builtin_module()]
print(data)
#[('/usr/lib/python3.5/os.py', 57, 25),
# ('/usr/lib/python3.5/os.py', 77, 22),
# ('/usr/lib/python3.5/os.py', 98, 22)]
```
Is not that simple to fix that at our level.

**Comment by DamnWidget** [comment]
@asfaltboy can you please give a try to branch [issue_497](https://github.com/DamnWidget/anaconda/tree/issue_497) ?

**Comment by asfaltboy** [comment]
@DamnWidget you are correct, I neglected to mention that in my case the conditional imports are of classes that share common ancestor class, which means that both "re-use" some methods; which leads to both being resolved to the same definition.

**Comment by DamnWidget** [comment]
I had to look in Google what LGTM means xDDDD (I am getting old), I will do a small release with the fix as soon as I fix the tests for Python 2.6 that are currently failing because the change :(

**Comment by asfaltboy** [comment]
Great, thanks for your help Oscar!
